web/src/components/network/NetworkPage.jsx

```diff
@@ -83,20 +85,56 @@ const NoWifiConnections = ({ wifiScanSupported, openWifiSelector }) => {
 export default function NetworkPage() {
   const { network: client } = useInstallerClient();
   const [connections, setConnections] = useState(undefined);
+  const [devices, setDevices] = useState(undefined);
   const [selectedConnection, setSelectedConnection] = useState(null);
   const [wifiScanSupported, setWifiScanSupported] = useState(false);
   const [wifiSelectorOpen, setWifiSelectorOpen] = useState(false);
 
   const openWifiSelector = () => setWifiSelectorOpen(true);
   const closeWifiSelector = () => setWifiSelectorOpen(false);
 
+  useEffect(() => {
+    return client.onNetworkChange(({ type, payload }) => {
+      switch (type) {
+        case NetworkEventTypes.DEVICE_ADDED: {
+          setDevices((devs) => {
+            const newDevices = devs.filter((d) => d.name !== payload.name);
```

Comment:
NP: in all(?) handling of NetworkEventTypes.DEVICE_FOOED, `newDevices` should read `oldDevices` or "existing" or "current", whatever, but definitely not "new".